tests/unit/test_kernel.py

```diff
@@ -212,22 +215,287 @@ def expected_divergence_x_grad_y(
+        k1.grad_x_elementwise.return_value = np.full(dimension, 1.0)
+        k1.grad_x.return_value = np.full((num_points, num_points, dimension), 1.0)
+        k1.grad_y_elementwise.return_value = np.full(dimension, 1.0)
+        k1.grad_y.return_value = np.full((num_points, num_points, dimension), 1.0)
+        k1.divergence_x_grad_y_elementwise.return_value = np.array(1.0)
+        k1.divergence_x_grad_y.return_value = np.full((num_points, num_points), 1.0)
+
+        k2 = MagicMock(spec=Kernel)
+        k2.compute_elementwise.return_value = np.array(2.0)
+        k2.compute.return_value = np.full((num_points, num_points), 2.0)
+        k2.grad_x_elementwise.return_value = np.full(dimension, 2.0)
+        k2.grad_x.return_value = np.full((num_points, num_points, dimension), 2.0)
+        k2.grad_y_elementwise.return_value = np.full(dimension, 2.0)
+        k2.grad_y.return_value = np.full((num_points, num_points, dimension), 2.0)
+        k2.divergence_x_grad_y_elementwise.return_value = np.array(2.0)
+        k2.divergence_x_grad_y.return_value = np.full((num_points, num_points), 2.0)
+
+        self.first_kernel = k1
+        self.second_kernel = k2
+
+    def to_additive_kernel(self) -> AdditiveKernel:
+        """Construct an Additive kernel."""
+        return AdditiveKernel(self.first_kernel, self.second_kernel)
+
+    def to_product_kernel(self) -> ProductKernel:
+        """Construct a Product kernel."""
+        return ProductKernel(self.first_kernel, self.second_kernel)
+
+
+class TestAdditiveKernel(
+    BaseKernelTest[AdditiveKernel],
+    KernelMeanTest[AdditiveKernel],
+    KernelGradientTest[AdditiveKernel],
+):
+    """Test ``coreax.kernel.AdditiveKernel``."""
+
+    # Set size and dimension of mock "dataset" that the mocked kernel will act on
+    mock_num_points = 5
+    mock_dimension = 3
+
+    @pytest.fixture(scope="class")
+    def kernel(self) -> AdditiveKernel:
+        """Return a mocked paired kernel function."""
+        return _MockedPairedKernel(
+            num_points=self.mock_num_points, dimension=self.mock_dimension
+        ).to_additive_kernel()
+
+    @pytest.fixture(params=["floats", "vectors", "arrays"])
+    def problem(self, request, kernel: AdditiveKernel) -> _Problem:
+        r"""
+        Test problems for the Additive kernel.
+
+        Given kernel functions :math:`k:\mathbb{R}^d \times \mathbb{R}^d \to \mathbb{R}`
+        and :math:`l:\mathbb{R}^d \times \mathbb{R}^d \to \mathbb{R}`, define the
+        additive kernel :math:`p:\mathbb{R}^d \times \mathbb{R}^d \to \mathbb{R}` where
+        :math:`p(x,y) := k(x,y) + l(x,y)`
+
+        We consider the simplest possible example of adding two Linear kernels together
+        with the following cases:
+        - `floats`: where x and y are floats
+        - `vectors`: where x and y are vectors of the same size
+        - `arrays`: where x and y are arrays of the same shape
+        """
+        mode = request.param
+        x = 0.5
```

Comment:
Very minor point, but is there a reason why these are defined outside of the if statement below for floats?

Reply:
Just following the structure of the other kernel tests already in, don't think there is any reason